filesetup.c

```diff
@@ -1254,6 +1343,44 @@ int setup_files(struct thread_data *td)
 		temp_stall_ts = 0;
 	}
 
+	if (err)
+		goto err_out;
+
+	/*
+	 * Prepopulate regular files with data. It might be expected
+	 * to read some "real" data instead of zero'ed files (if no writes
+	 * to file occurred prior to a read job).
+	 */
+	if (td->o.prepopulate && td_read(td)) {
```

Comment:
I wouldn't optimize only for reads unless your going to introduce an additional state (which would be kinda awesome because it opens the opportunity to use this to entirely replace the existing code). There can be reasons why you want to force pre-population when you're doing writes (e.g. job stops part way through and then reads are done and the reads perform faster than expected).

Reply:
I see and agree. Done.